rse-cmdline-git.Rmd

```diff
@@ -0,0 +1,2296 @@
+# Git at the command line {#rse-cmdline-git}
+
+## Questions {#rse-cmdline-git-questions}
+
+```{r, child="questions/rse-cmdline-git.md"}
+```
+
+## Introduction {#rse-cmdline-git-intro}
+
+In a previous lesson (TODO: provide link),
+we learned the basics of Git using a graphical interface (GitKraken or R Studio).
+Like many computer programs,
+using Git at the command line is a potentially more powerful option,
+as it allows you to access all the features.
+This chapter describes how the basic "add, commit, push, pull" cycle works at the command line,
+before introducing new concepts including branching and pull requests.
+
+
+## Setting up Git {#rse-cmdline-git-setup}
+
+When we use Git at the command line for the first time,
+we need to configure a few things. 
+
+On a command line, Git commands are written as `git verb options`,
+where `verb` is what we actually want to do
+and `options` is additional optional information which may or may not be required for the `verb`.
+Here is how Dracula sets up his new laptop:
+
+```shell
+$ git config --global user.name "Vlad Dracula"
+$ git config --global user.email "vlad@tran.sylvan.ia"
+```
+
+(Please use your own name and email address instead of Dracula's.)
+
+For these lessons we will be interacting with [GitHub](https://github.com/),
+so the email address should be the same as the one used when setting up your GitHub account.
+
+The commands we just ran above only need to be run once: the flag `--global` tells Git
+to use the settings for every project, in your user account, on this computer.
+You can re-run them at any time if you want to change your details.
+
+You can also check your settings at any time:
+
+```shell
+$ git config --list
+```
+
+> **Git Help and Manual**
+>
+> Always remember that if you forget a `git` command,
+> you can access the list of commands by using `-h` and access the Git manual by using `--help` :
+>
+> ```shell
+> $ git config -h
+> $ git config --help
+> ```
+
+
+## Creating a repository {#rse-cmdline-git-repos}
+
+Once Git is configured,
+we can start using it.
+
+We will continue with the story of Wolfman and Dracula who are investigating if it
```

Comment:
Do we want to stick to a narrative that aligns with the general outcome of using Git for a software package? Rather than Wolfman and Dracula and the lander?

Reply:
This applies to all the below text too, including "mkdir planets" etc. I think it would be better to be aligned with our goal of using Git for making software packages.